app/src/main/java/com/picone/go4lunch/presentation/ui/MainActivity.java

```diff
@@ -1,29 +1,113 @@
 package com.picone.go4lunch.presentation.ui;
 
 import android.os.Bundle;
+import android.view.View;
 
 import androidx.appcompat.app.AppCompatActivity;
-import androidx.navigation.fragment.NavHostFragment;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+import androidx.navigation.ui.AppBarConfiguration;
 import androidx.navigation.ui.NavigationUI;
 
+import com.facebook.login.LoginManager;
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.firebase.auth.FirebaseAuth;
 import com.picone.go4lunch.R;
 import com.picone.go4lunch.databinding.ActivityMainBinding;
+import com.picone.go4lunch.presentation.viewModels.LoginViewModel;
 
 public class MainActivity extends AppCompatActivity {
 
+    public static LoginViewModel LOGIN_VIEW_MODEL;
+
     private ActivityMainBinding mBinding;
+    private NavController mNavController;
```

Comment:
make mGoogleSignInClient public and use it in BaseFragment instead of creating a new instance of it.

Reply:
done